### PR TITLE
Create Pending Changelog Checker Action and initial CHANGELOG.md and CHANGELOG-PENDING.md files

### LINKED ISSUE #246: automated CHANGELOG.md check for celestia-node
There should be an automated step that ensures that a CHANGELOG.md file is appropriately updated for all merges to the `main` branch.

There should be a proper description in an issue and corresponding PR as well as a summary in the CHANGELOG.md file. This will be enforced through an automated action. In rare cases for small PRs it may be acceptable to skip this.

### PR DESCRIPTION
**Pull request by jbowen93**
Create `CHANGELOG.md` and `CHANGELOG-PENDING.md` files and a corresponding "Pending Changelog Checker" action that runs on all PRs to the `main` branch. 

This check will only pass if the `CHANGELOG-PENDING.md` file is updated in the PR. 

This check can be bypassed by labeling the PR with `no_changelog`

Resolves #246

### CODE REVIEW

File: CHANGELOG.md
Comment (liamsi) [inline CHANGELOG.md:21]: Just to clarify: the workflow here will be to populate the CHANGELOG-PENDING.md on each PR. Then, before a release gets cut, the content of the CHANGELOG-PENDING.md gets copied over (or rather cut & pasted) to the CHANGELOG.md. 

Can we cut a post-devnet release right after this here gets merged? After https://github.com/celestiaorg/celestia-node/issues/226 we can do a small patch release. (cc @renaynay @Wondertan).

File: CHANGELOG.md
Comment (liamsi) [inline CHANGELOG.md:3]:
```suggestion
## vX.Y.Z
```

File: CHANGELOG-PENDING.md
Comment (liamsi) [inline CHANGELOG-PENDING.md:3]:
```suggestion
## vX.Y.Z
```

File: CHANGELOG.md
Comment (jbowen93) [reply CHANGELOG.md:21]: Correct, I'm planning to have a similar action to ensure the CHANGELOG.md is updated during manual releases. The manual release process is TBD.

**Review by renaynay — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
Thanks for this! LGTM (after addressing Ismail's comments)

**Review by renaynay — APPROVED** [review]

**Review by liamsi — APPROVED** [review]
great, thanks! Let's cut the first release 🚀

**Pull request merged**
